cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -24,11 +24,110 @@ use std::collections::HashSet;
 type Hash = H256;
 type Block = crate::ethereum::Block<Transaction>;
 
-pub async fn matching_transaction<C>(
+pub async fn matching_deployed_transaction<'b, C>(
+    blockchain_connector: C,
+    start_of_swap: NaiveDateTime,
+    _bytecode: Bytes,
+    _create_contract: CreateContract,
+) -> anyhow::Result<&'b Transaction>
+where
+    C: LatestBlock<Block = Option<Block>>
+        + BlockByHash<Block = Option<Block>, BlockHash = Hash>
+        + ReceiptByHash<Receipt = Option<TransactionReceipt>, TransactionHash = Hash>
+        + Clone,
+{
+    let (transaction, _) =
+        matching_transaction(blockchain_connector, start_of_swap, |_transaction| {
+            // TODO: Add logic for matching a deployed transaction.
+            (false, None)
+        })
+        .await?;
+
+    Ok(transaction)
+}
+
+pub async fn matching_funded_transaction<'b, C>(
```

Comment:
Bear in mind that funding for erc20 is different than for ether :)

The erc20 one needs to look for a Transfer event where as the ether one only needs to look for a transaction.

Reply:
Yep, I'm across that, will however keep it mind.  Thanks